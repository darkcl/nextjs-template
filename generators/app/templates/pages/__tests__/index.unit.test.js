import React from 'react';
import toJson from 'enzyme-to-json';
import { render } from 'enzyme';

import ApiClient from '../../lib/api-client';
import Index from '../index';

const apiClient = new ApiClient('.');

describe('<Index />', () => {
  it('getInitialProps should return object with status code', async () => {
    const props = await Index.getInitialProps();

    expect(props).toBeDefined();
    expect(props.data).toHaveLength(10);
    expect(props.statusCode).toBe(200);
  });

  it('should render correctly with searchable result', async () => {
    const res = await apiClient.getShows('batman');

    const index = render(<Index {...res} />);
    expect(toJson(index)).toMatchSnapshot();
  });

  it('should render correctly with non-searchable result', async () => {
    const res = await apiClient.getShows('helloword');

    const index = render(<Index {...res} />);
    expect(toJson(index)).toMatchSnapshot();
  });
});
