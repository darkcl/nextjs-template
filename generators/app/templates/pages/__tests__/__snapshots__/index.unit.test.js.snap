// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index /> should render correctly with non-searchable result 1`] = `
Array [
  <div
    class="header"
  >
    <div
      class="container"
    >
      <a
        href="/"
      >
        <img
          src="undefined/static/47cd76e43f74bbc2e1baaf194d07e1fa/images/favicon.png"
        />
      </a>
      <span>
        header.title
      </span>
    </div>
  </div>,
  <div
    class="container"
  >
    <p>
      An error 404 occurred on server
    </p>
  </div>,
  <div
    class="footer"
  >
    <div
      class="container"
    >
      footer.text
    </div>
  </div>,
]
`;

exports[`<Index /> should render correctly with searchable result 1`] = `
Array [
  <div
    class="header"
  >
    <div
      class="container"
    >
      <a
        href="/"
      >
        <img
          src="undefined/static/47cd76e43f74bbc2e1baaf194d07e1fa/images/favicon.png"
        />
      </a>
      <span>
        header.title
      </span>
    </div>
  </div>,
  <div
    class="container"
  >
    <div
      class="index"
    >
      <h1>
        content.header
      </h1>
      <p>
        content.subTitle
      </p>
      <ul>
        <li>
          <a
            href="/shows/975"
          >
            Batman (8.2)
          </a>
        </li>
        <li>
          <a
            href="/shows/481"
          >
            The Batman (8.3)
          </a>
        </li>
        <li>
          <a
            href="/shows/504"
          >
            Batman Beyond (9)
          </a>
        </li>
        <li>
          <a
            href="/shows/757"
          >
            Batman: The Animated Series (9.3)
          </a>
        </li>
        <li>
          <a
            href="/shows/11464"
          >
            Batman Unlimited (10)
          </a>
        </li>
        <li>
          <a
            href="/shows/3557"
          >
            Beware the Batman (10)
          </a>
        </li>
        <li>
          <a
            href="/shows/33618"
          >
            The Adventures of Batman 
          </a>
        </li>
        <li>
          <a
            href="/shows/900"
          >
            Batman: The Brave and The Bold (9.4)
          </a>
        </li>
        <li>
          <a
            href="/shows/5951"
          >
            The New Batman Adventures (8.6)
          </a>
        </li>
        <li>
          <a
            href="/shows/22309"
          >
            Batman: Black and White 
          </a>
        </li>
      </ul>
    </div>
  </div>,
  <div
    class="footer"
  >
    <div
      class="container"
    >
      footer.text
    </div>
  </div>,
]
`;
